controllers/chat/getSignedChannelList.js

```diff
@@ -0,0 +1,78 @@
+const {responseSuccess} = require('../../utils/Responses');
+const GetstreamSingleton = require('../../vendor/getstream/singleton');
+
+const MAX_ITERATIONS = 10;
+
+const __queryChannelBuilder = async (userId, limit, offset) => {
+  const client = GetstreamSingleton.getChatInstance();
+  const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
+    limit: limit,
+    offset: offset,
+    state: true
+  });
+
+  return channels;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `__queryChannelBuilder` function is making a call to an external service. It would be better to handle potential errors that might occur during this operation. This could be done by wrapping the `await client.queryChannels` call in a try-catch block.

```diff
const __queryChannelBuilder = async (userId, limit, offset) => {
  const client = GetstreamSingleton.getChatInstance();
  try {
    const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
      limit: limit,
      offset: offset,
      state: true
    });
    return channels;
  } catch (error) {
    console.error(`Error querying channels for user ${userId}: `, error);
    throw error;
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->